ibm/data_source_ibm_cr_namespaces.go

```diff
@@ -54,38 +81,42 @@ func dataIBMContainerRegistryNamespaces() *schema.Resource {
 	}
 }
 
-func dataIBMContainerRegistryNamespacesRead(d *schema.ResourceData, meta interface{}) error {
-	userDetails, err := meta.(ClientSession).BluemixUserDetails()
-	if err != nil {
-		return err
-	}
-	accountID := userDetails.userAccount
-
-	crClient, err := meta.(ClientSession).ContainerRegistryAPI()
+func dataIBMContainerRegistryNamespacesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	containerRegistryClient, err := meta.(ClientSession).ContainerRegistryV1()
 	if err != nil {
-		return err
-	}
-	target := registryv1.NamespaceTargetHeader{
-		AccountID: accountID,
+		return diag.FromErr(err)
 	}
 
-	crAPI := crClient.Namespaces()
+	listNamespaceDetailsOptions := &containerregistryv1.ListNamespaceDetailsOptions{}
```

Comment:
The code releated/ to set the accoutnID is removed from options...Will this effect by any chance if user uses their serviceID API key

Reply:
The AccountID is a "global parameter" -- it is set in [`ibm/config.go`](https://github.com/IBM-Cloud/terraform-provider-ibm/pull/2372/files#diff-6dc66490d06e928e5da63fa4d3ce2c6410a6d6bdacb2087026546d583b8c5527R1280) as part of the initialization of the Container Registry SDK.

Reply:
What Mike said....